apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -152,3 +171,68 @@ export const Contacts = {
 		return contactId;
 	},
 };
+
+export async function createContact(params: CreateContactParams): Promise<string> {
+	const { name, emails, phones, customFields = {}, contactManager, channels, unknown } = params;
+
+	if (contactManager) {
+		const contactManagerUser = await Users.findOneById(contactManager, { projection: { roles: 1 } });
+		if (!contactManagerUser) {
+			throw new Error('error-contact-manager-not-found');
+		}
+		if (!contactManagerUser.roles || !Array.isArray(contactManagerUser.roles) || !contactManagerUser.roles.includes('livechat-agent')) {
+			throw new Error('error-invalid-contact-manager');
+		}
+	}
+
+	const allowedCustomFields = await getAllowedCustomFields();
+	validateCustomFields(allowedCustomFields, customFields);
+
+	const { insertedId } = await LivechatContacts.insertOne({
```

Comment:
IF this will be the "new" way of adding visitors/contacts, shouldn't we put a note on the visitors model so people is aware? 👀 (not sure if that's gonna happen on another PR)

Reply:
Visitors and contacts are different things. Visitors will be created the same way they're today. This introduces the capability of creating the new entity called Contact

Reply:
Interesting 🤔 what's the diff between them?

Reply:
Contact is a new entity being introduced by the Single Contact ID project. The main idea is to have one entity that centralizes all the information of all channels that a customer can contact. It will "group" a bunch of visitors who represent the same person and allow the workspace to verify the identity of this contact. We can schedule a meeting with @renatobecker  if you want to know more about the project and its objectives.